src/rules/default-import-match-filename.js

```diff
@@ -0,0 +1,206 @@
+import docsUrl from '../docsUrl'
+import isStaticRequire from '../core/staticRequire'
+import Path from 'path'
+import fs from 'fs'
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function removeExtension(filename) {
+  return Path.basename(filename, Path.extname(filename))
+}
+
+/**
+ * @param {string} filename
+ * @returns {string}
+ */
+function normalizeFilename(filename) {
+  return filename.replace(/[-_.]/g, '').toLowerCase()
+}
+
+/**
+ * Test if local name matches filename.
+ * @param {string} localName
+ * @param {string} filename
+ * @returns {boolean}
+ */
+function isCompatible(localName, filename) {
+  const normalizedLocalName = localName.replace(/_/g, '').toLowerCase()
+
+  return (
+    normalizedLocalName === normalizeFilename(filename) ||
+    normalizedLocalName === normalizeFilename(removeExtension(filename))
+  )
+}
+
+/**
+ * Match 'foo' but not 'foo/bar.js' and './foo'
+ * @param {string} path
+ * @returns {boolean}
+ */
+function isBarePackageImport(path) {
+  return path !== '.' && path !== '..' && !path.includes('/')
+}
+
+/**
+ * Match paths consisting of only '.' and '..', like '.', './', '..', '../..'.
+ * @param {string} path
+ * @returns {boolean}
+ */
+function isAncestorRelativePath(path) {
+  return (
+    path.length > 0 &&
+    path[0] !== '/' &&
+    path
+      .split('/')
```

Comment:
```suggestion
      .split(/[/\\]/)
```

Reply:
Is this needed? I have never seen people using backslash in an import path.

Reply:
I'd assume Windows users might?

Reply:
its allowed thus itshould be supported, normally no one should do this as its making code work only on windows machines

Reply:
Arguably that's true for absolute linux paths, just in reverse.